pinot-tools/src/main/java/org/apache/pinot/tools/MultiDirQuickstart.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.tools;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.tools.admin.PinotAdministrator;
+import org.apache.pinot.tools.admin.command.QuickstartRunner;
+
+
+/**
+ * A Quickstart to demo how multi directory support works and its typical configs.
+ */
+public class MultiDirQuickstart extends Quickstart {
+  @Override
+  public List<String> types() {
+    return Arrays.asList("OFFLINE_MULTIDIR", "OFFLINE-MULTIDIR", "BATCH_MULTIDIR", "BATCH-MULTIDIR");
+  }
+
+  @Override
+  protected Map<String, Object> getConfigOverrides() {
+    Map<String, Object> properties = new HashMap<>(super.getConfigOverrides());
+    properties.put("pinot.server.instance.segment.directory.loader", "tierBased");
+    properties.put("controller.segment.relocator.frequencyPeriod", "60s");
+    properties.put("controller.segmentRelocator.initialDelayInSeconds", "10");
+    properties.put("controller.segmentRelocator.enableLocalTierMigration", "true");
+    /*
+     * One can also set `dataDir` as part of tierConfigs in TableConfig to overwrite the instance configs (or set as
+     * cluster configs), but it's recommended to use instance (or cluster) configs for consistency across tables.
+     * "tierBackendProperties": {
+     *     "dataDir": "/tmp/multidir_test/hotTier"
+     *  }
+     */
+    properties.put("pinot.server.instance.tierConfigs.0.tierName", "hotTier");
+    properties.put("pinot.server.instance.tierConfigs.0.dataDir", "/tmp/multidir_test/hotTier");
+    properties.put("pinot.server.instance.tierConfigs.1.tierName", "coldTier");
+    properties.put("pinot.server.instance.tierConfigs.1.dataDir", "/tmp/multidir_test/coldTier");
```

Comment:
looks like this "0" and "1" can be anything at all, and is being used just to distinguish the keys. would it be better to have this instead? this is also how we do it in FS configs I think
```
pinot.server.instance.tierConfigs.tierNames=hotTier,coldTier
pinot.server.instance.tierConfigs.hotTier.dataDir=/tmp/multidir_test/hotTier
pinot.server.instance.tierConfigs.coldTier.dataDir=/tmp/multidir_test/coldTier
```

Reply:
right, the part of "0" or "1" can be anything unique.

I think I saw both conventions in pinot configs. I chose this one to avoid worries about escaping name separator in tier names, which should rare anyway. Will check out FS config.